generation.go

```diff
@@ -0,0 +1,388 @@
+package gen
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+type Generation struct {
+	api *API
+
+	enums     []*Enum
+	functions []*Function
+	structs   []*Struct
+
+	lookupEnum        map[string]*Enum
+	lookupNonTypedefs map[string]string
+	lookupStruct      map[string]*Struct
+}
+
+func newGeneration(a *API, headerFiles []*HeaderFile) *Generation {
+	gen := &Generation{
+		api: a,
+
+		lookupEnum:        map[string]*Enum{},
+		lookupNonTypedefs: map[string]string{},
+		lookupStruct: map[string]*Struct{
+			"cxstring": &Struct{
+				Name:  "cxstring",
+				CName: "CXString",
+			},
+		},
+	}
+	gen.init(headerFiles)
+
+	return gen
+}
+
+func (g *Generation) init(headerFiles []*HeaderFile) {
+	for _, h := range headerFiles {
```

Comment:
1) yepp
2) I like the idea of its own struct, i have to tinker a little bit with that